contracts/vesting/src/tests/instantiate.rs

```diff
@@ -0,0 +1,274 @@
+
+    let vesting_token = VestingTokenInfo {
+        name: String::from_str(&env, "Phoenix"),
+        symbol: String::from_str(&env, "PHO"),
+        decimals: 6,
+        address: Address::generate(&env),
+        total_supply: 480,
+    };
+    let vesting_balances = vec![
+        &env,
+        VestingBalance {
+            address: vester1,
+            balance: 240,
+            curve: Curve::SaturatingLinear(SaturatingLinear {
+                min_x: 15,
+                min_y: 120,
+                max_x: 60,
+                max_y: 0,
+            }),
+        },
+        VestingBalance {
+            address: vester2,
+            balance: 240,
+            curve: Curve::SaturatingLinear(SaturatingLinear {
+                min_x: 30,
+                min_y: 240,
+                max_x: 120,
+                max_y: 0,
+            }),
+        },
+    ];
+
+    let vesting_client = instantiate_vesting_client(&env);
+    env.ledger().with_mut(|li| li.timestamp = 1000);
+    vesting_client.initialize(
+        &admin,
+        &vesting_token,
+        &vesting_balances,
+        &None,
+        &None,
+        &10u32,
+    );
+
+    assert_eq!(vesting_client.query_token_info(), vesting_token);
+    assert_eq!(
+        vesting_client.query_vesting_whitelist(),
+        vec![&env, admin.clone()]
+    );
+}
+
+#[should_panic(expected = "Vesting: Initialize: At least one balance must be provided.")]
+#[test]
+fn instantiate_contract_without_any_vesting_balances_should_fail() {
+    let env = Env::default();
+    env.mock_all_auths();
+
+    let admin = Address::generate(&env);
+    let whitelisted_account = Address::generate(&env);
+
+    let vesting_token = VestingTokenInfo {
+        name: String::from_str(&env, "Phoenix"),
+        symbol: String::from_str(&env, "PHO"),
+        decimals: 6,
+        address: Address::generate(&env),
+        total_supply: 0,
+    };
+    let vesting_balances = vec![&env];
+
+    let allowed_vesters = vec![&env, whitelisted_account.clone()];
+
+    let vesting_client = instantiate_vesting_client(&env);
+    env.ledger().with_mut(|li| li.timestamp = 1000);
+    vesting_client.initialize(
+        &admin,
+        &vesting_token,
+        &vesting_balances,
+        &None,
+        &Some(allowed_vesters),
+        &10u32,
+    );
+}
+
+#[should_panic(expected = "Vesting: Initialize: total supply over the cap")]
+#[test]
+fn instantiate_contract_should_panic_when_supply_over_the_cap() {
+    let env = Env::default();
+    env.mock_all_auths();
+
+    let admin = Address::generate(&env);
+    let vester1 = Address::generate(&env);
+    let vester2 = Address::generate(&env);
+    let whitelisted_account = Address::generate(&env);
+
+    let vesting_token = VestingTokenInfo {
+        name: String::from_str(&env, "Phoenix"),
+        symbol: String::from_str(&env, "PHO"),
+        decimals: 6,
+        address: Address::generate(&env),
+        total_supply: 0,
+    };
```

Comment:
What fails in this test is that the `total_supply` is 0 and then you try to initialize it with anything above?
Please, clean up this test, remove what's not needed here and adjust those values to anything that could be a real situation.

Reply:
`total_supply` comes from `vesting_balances` and the `capacity` of the minter. Skimmed as much as possible from the test. Please, let me know if I missed something